plugins/ci/skills/oc-auth/SKILL.md

```diff
@@ -0,0 +1,188 @@
+---
+name: OC Authentication Helper
+description: Helper skill to retrieve OAuth tokens from the correct OpenShift cluster context when multiple clusters are configured
+---
+
+# OC Authentication Helper
+
+This skill provides a centralized way to retrieve OAuth tokens from specific OpenShift clusters when multiple cluster contexts are configured in the user's kubeconfig.
+
+## When to Use This Skill
+
+Use this skill whenever you need to:
+- Get an OAuth token for API authentication from a specific OpenShift cluster
+- Verify authentication to a specific cluster
+- Work with multiple OpenShift cluster contexts simultaneously
+
+This skill is used by all commands that need to authenticate with OpenShift clusters:
+- `ask-sippy` command (DPCR cluster)
+- `trigger-periodic`, `trigger-postsubmit`, `trigger-presubmit` commands (app.ci cluster)
+- `query-job-status` command (app.ci cluster)
+
+The skill provides a single `curl_with_token.sh` script that wraps curl and automatically handles OAuth token retrieval and injection, preventing accidental token exposure.
+
+## Prerequisites
+
+1. **oc CLI Installation**
+   - Check if installed: `which oc`
+   - If not installed, provide instructions for the user's platform
+   - Installation guide: https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/getting-started-cli.html
+
+2. **User Authentication**
+   - User must be logged in to the target cluster via browser-based authentication
+   - Each `oc login` creates a new context in the kubeconfig
+
+## How It Works
+
+The `oc` CLI maintains multiple cluster contexts in `~/.kube/config`. When a user runs `oc login` to different clusters, each login creates a separate context. This skill:
+
+1. Lists all available contexts
+2. Searches for the context matching the target cluster by cluster name pattern
+3. Retrieves the OAuth token from that specific context
+4. Returns the token for use in API calls
+
+## Cluster Identifiers
+
+The skill supports two cluster identifiers:
+
+### 1. `app.ci` - OpenShift CI Cluster
+- **Cluster name pattern**: `ci-l2s4-p1`
+- **Console URL**: https://console-openshift-console.apps.ci.l2s4.p1.openshiftapps.com/
+- **API Server**: https://api.ci.l2s4.p1.openshiftapps.com:6443
+- **Used by**: trigger-periodic, trigger-postsubmit, trigger-presubmit, query-job-status
+
+### 2. `dpcr` - DPCR Cluster
+- **Cluster name pattern**: `cr-j7t7-p1`
+- **Console URL**: https://console-openshift-console.apps.cr.j7t7.p1.openshiftapps.com/
+- **API Server**: https://api.cr.j7t7.p1.openshiftapps.com:6443
+- **Used by**: ask-sippy
+
```

Comment:
maybe there could be another section here which makes it so any Openshift cluster can be used given the URL? so folks don't have to add every possible cluster URL to this skill doc.